modules/microshift-storage-persistent-storage-pv.adoc

```diff
@@ -0,0 +1,281 @@
+have the same set of access modes, either of them can match a claim with
+those modes. There is no ordering between types of volumes and no way to
+choose one type over another.
+
+All volumes with the same modes are grouped, and then sorted by size,
+smallest to largest. The binder gets the group with matching modes and
+iterates over each, in size order, until one size matches.
+
+The following table lists the access modes:
+
+.Access modes
+[cols="1,1,3",options="header"]
+|===
+|Access Mode |CLI abbreviation |Description
+|ReadWriteOnce
+|`RWO`
+|The volume can be mounted as read-write by a single node.
+ifdef::openshift-enterprise,openshift-webscale,openshift-origin[]
+|ReadOnlyMany
+|`ROX`
+|The volume can be mounted as read-only by many nodes.
+|ReadWriteMany
+|`RWX`
+|The volume can be mounted as read-write by many nodes.
+endif::[]
+|===
+
+[IMPORTANT]
+====
+Volume access modes are descriptors of volume capabilities. They
+are not enforced constraints. The storage provider is responsible for
+runtime errors resulting from invalid use of the resource.
+
+For example, NFS offers `ReadWriteOnce` access mode. You must
+mark the claims as `read-only` if you want to use the volume's
+ROX capability. Errors in the provider show up at runtime as mount errors.
+
+iSCSI and Fibre Channel volumes do not currently have any fencing
+mechanisms. You must ensure the volumes are only used by one node at a
+time. In certain situations, such as draining a node, the volumes can be
+used simultaneously by two nodes. Before draining the node, first ensure
+the pods that use these volumes are deleted.
+====
+
+.Supported access modes for PVs
+[cols=",^v,^v,^v", width="100%",options="header"]
+|===
+|Volume plug-in  |ReadWriteOnce ^[1]^  |ReadOnlyMany  |ReadWriteMany
+|AWS EBS ^[2]^ | ✅ | - |  -
+ifdef::openshift-enterprise,openshift-webscale,openshift-origin[]
+|Azure File | ✅ | ✅ | ✅
+|Azure Disk | ✅ | - | -
+//|Ceph RBD  | ✅ | ✅ |  -
+//|CephFS  | ✅ | ✅ |  ✅
+|Cinder  | ✅ | - |  -
+|Fibre Channel  | ✅ | ✅ |  -
+|GCE Persistent Disk  | ✅ | - |  -
+//|GlusterFS  | ✅ | ✅ | ✅
+|HostPath  | ✅ | - |  -
+|iSCSI  | ✅ | ✅ |  -
+|Local volume | ✅ | - |  -
+|NFS  | ✅ | ✅ | ✅
+|OpenStack Manila  | - | - | ✅
+|{rh-storage-first}  | ✅ | - | ✅
+|VMware vSphere | ✅ | - | ✅ ^[3]^ 
+endif::[]
+
+|===
+[.small]
+--
+1. ReadWriteOnce (RWO) volumes cannot be mounted on multiple nodes. If a node fails, the system does not allow the attached RWO volume to be mounted on a new node because it is already assigned to the failed node. If you encounter a multi-attach error message as a result, force delete the pod on a shutdown or crashed node to avoid data loss in critical workloads, such as when dynamic persistent volumes are attached.
+2. Use a recreate deployment strategy for pods that rely on AWS EBS.
+3. If the underlying vSphere environment supports the vSAN file service, then the vSphere Container Storage Interface (CSI) Driver Operator installed by
```

Comment:
vSphere and the vSAN file service aren't supported either.  We could drop this whole paragraph.